custom/icds_reports/reports/ucr.py

```diff
@@ -339,5 +339,96 @@ def get_total_records(self):
             return session.connection().execute(count_query, **self.helper.sql_alchemy_filter_values).scalar()
 
 
+CcsRecordMonthlyViewAlchemy = sqlalchemy.Table(
+    'ccs_record_monthly_view', metadata, autoload=True
+)
+
+
+class CcsRecordMonthlyUCR(ConfigurableReportCustomQueryProvider):
+    def __init__(self, report_data_source):
+        self.report_data_source = report_data_source
+        self.helper = self.report_data_source.helper
+        self.helper.set_table(CcsRecordMonthlyViewAlchemy)
+
+    @property
+    def table(self):
+        return CcsRecordMonthlyViewAlchemy
+
+    def _get_query_object(self, total_row=False):
+
+        def format_column(title, logic):
+            return func.count(self.table.c.case_id).filter(logic).label(title)
+
+        pregnant = self.table.c.pregnant == 1
+        lactating = self.table.c.lactating == 1
+        st_caste = self.table.c.caste == 'st'
+        sc_caste = self.table.c.caste == 'sc'
+        rations_gte_21 = self.table.c.num_rations_distributed >= 21
+        rations_lt_21 = self.table.c.num_rations_distributed < 21
+        disabled = self.table.c.disabled == 'yes'
+        minority = self.table.c.minority == 'yes'
+        rations_none = self.table.c.num_rations_distributed == 0
```

Comment:
I think you also want to check if this column is NULL (i'm not sure if that's possible in the current schema but it would be good to add since the check was in the previous data source definition)

Reply:
Forgot to reply to this when I made the fix, but: https://github.com/dimagi/commcare-hq/pull/23552/commits/1e0eacb1663a5945b9145bd2cc80d7effc14449c